src/util/time.h

```diff
@@ -14,22 +14,37 @@
 
 using namespace std::chrono_literals;
 
+/** Mockable clock in the context of tests, otherwise the system clock */
+struct NodeClock : std::chrono::system_clock {
+    using time_point = std::chrono::time_point<NodeClock>;
+    /** Return current system time or mocked time, if set */
+    static time_point now() noexcept;
+    static std::time_t to_time_t(const time_point&) = delete; // unused
+    static time_point from_time_t(std::time_t) = delete;      // unused
+};
+using NodeSeconds = std::chrono::time_point<NodeClock, std::chrono::seconds>;
+
 using SteadySeconds = std::chrono::time_point<std::chrono::steady_clock, std::chrono::seconds>;
 using SteadyMilliseconds = std::chrono::time_point<std::chrono::steady_clock, std::chrono::milliseconds>;
 using SteadyMicroseconds = std::chrono::time_point<std::chrono::steady_clock, std::chrono::microseconds>;
 
 void UninterruptibleSleep(const std::chrono::microseconds& n);
 
 /**
- * Helper to count the seconds of a duration.
+ * Helper to count the seconds of a duration/time_point.
  *
- * All durations should be using std::chrono and calling this should generally
+ * All durations/time_points should be using std::chrono and calling this should generally
  * be avoided in code. Though, it is still preferred to an inline t.count() to
  * protect against a reliance on the exact type of t.
  *
- * This helper is used to convert durations before passing them over an
+ * This helper is used to convert durations/time_points before passing them over an
  * interface that doesn't support std::chrono (e.g. RPC, debug log, or the GUI)
  */
+template <typename Clock>
+constexpr int64_t count_seconds(std::chrono::time_point<Clock, std::chrono::seconds> t)
```

Comment:
Should this be called `seconds_since_epoch` (or the camelcase equivalent) perhaps? Looking back through #24697 it doesn't seem like it's just a temporary that will go away, so might be worth clearly distinguishing something that's working with a timepoint vs a duration.

At least as far as #24697 is concerned, this is effectively implementing  the behaviour of `NodeClock::to_time_t` so it's a bit funny that you explicitly deleted it above then reimplemented it here :)

Reply:
Is `time_t` using more than 32-bits on all platforms? Also rounding/truncation is implementation defined. So it seems best to avoid.

Also, I expect count_milliseconds to be added in the future, which `to_time_t` can't handle either.

`seconds_since_epoch` sounds good, will change.